### PR TITLE
ON-547: OriumSftMarketplace core functions

### PR DESCRIPTION
**Pull request by karacurt**

### CODE REVIEW

File: contracts/OriumSftMarketplace.sol [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    }
+
+    /** ============================ Core Functions  ================================== **/
+
+    /** ######### Setters ########### **/
+
+    /**
+     * @notice Sets the roles registry.
+     * @dev Only owner can set the roles registry.
+     */
```
Comment (ernanirst) [inline contracts/OriumSftMarketplace.sol:108]: The comment does not match what the function does

File: contracts/OriumSftMarketplace.sol [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+     * @param _treasury The address where the fees will be sent. If the treasury is address(0), the fees will be burned.
+     */
+    function setRoyaltyInfo(address _tokenAddress, uint256 _royaltyPercentageInWei, address _treasury) external {
+        require(
+            msg.sender == royaltyInfo[_tokenAddress].creator,
+            "OriumSftMarketplace: Only creator can set royalty info"
+        );
+
+        _setRoyalty(msg.sender, _tokenAddress, _royaltyPercentageInWei, _treasury);
+    }
```
Comment (ernanirst) [inline contracts/OriumSftMarketplace.sol:167]: Maybe let's combine these two functions and allow both the Contract Owner and the NFT Creator to call it. I see cases where users ask Orium to set the royalty, and others where users will want to set it themselves

File: test/OriumSftMarketplace.test.ts [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+  })
+
+  beforeEach(async () => {
+    // we are disabling this rule so ; may not be added automatically by prettier at the beginning of the line
+    // prettier-ignore
+    [marketplace, rolesRegistry, mockERC1155] = await loadFixture(deploySftMarketplaceContracts)
+  })
+
+  describe('Main Functions', async () => {
+    describe('Core Functions', async () => {
```
Comment (ernanirst) [inline test/OriumSftMarketplace.test.ts:44]: Across the test files, break the lines between `describe` and `it` statements. It's very hard to read when everything is combined like this

**Review by ernanirst — CHANGES_REQUESTED** [review]

File: test/OriumSftMarketplace.test.ts (hunk 3 above)
Comment (karacurt) [reply test/OriumSftMarketplace.test.ts:44]: i tried, but prettier/lint still deleting the space between those two

**Review by ernanirst — APPROVED** [review]

**Pull request merged**